### PR TITLE
Add formal EBNF spec

### PR DESCRIPTION
**Pull request by ilslv**
# Description

This PR adds a formal specification for `Cucumber Expressions` language. For more context see #41

### CODE REVIEW

File: ARCHITECTURE.md [hunk 1]
```diff
@@ -6,6 +6,35 @@ This document describes the grammar and production rules of Cucumber Expressions
 
 A Cucumber Expression has the following [EBNF](https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form) grammar:
 
+```ebnf
```
Comment (aslakhellesoy) [inline ARCHITECTURE.md:9]: Rather than adding a new EBNF grammar, can you please make modifications to the existing one? There should only be one grammar.

**Review by aslakhellesoy — CHANGES_REQUESTED** [review]

File: ARCHITECTURE.md (hunk 1 above)
Comment (aslakhellesoy) [reply ARCHITECTURE.md:9]: Also, it's important that our implementations are consistent with the formal grammar, so if we're going to update it with your proposed changes, we may also need to change the existing implementations.

Comment (ilslv) [reply ARCHITECTURE.md:9]: @aslakhellesoy 

> can you please make modifications to the existing one

existing grammar is context-specific and can't be called EBNF at all. Also it describes superset of `Cucumber Expression` while there is no need to do it. 
Old specification if left to provide insights into current implementation.

**Comment by mpkorstanje** [comment]
I couldn't find the time last year to do this, I reckon I won't find the time this year either.

**Comment by aslakhellesoy** [comment]
Hi @ilslv,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Pull request merged**